tests/phpunit/integration/Core/Authentication/Google_ProxyTest.php

```diff
@@ -43,6 +44,121 @@ public function test_get_site_fields() {
 		);
 	}
 
+	public function test_fetch_site_fields() {
+		$context      = new Context( GOOGLESITEKIT_PLUGIN_MAIN_FILE );
+		$google_proxy = new Google_Proxy( $context );
+		$credentials  = new Credentials( new Options( $context ) );
+
+		$this->fake_proxy_site_connection();
+
+		$pre_args = null;
+		$pre_url  = null;
+
+		// Use pre_http_request for backwards compatibility as http_api_debug is not fired for blocked requests before WP 5.3
+		add_filter(
+			'pre_http_request',
+			function ( $false, $args, $url ) use ( &$pre_args, &$pre_url ) {
+				$pre_args = $args;
+				$pre_url  = $url;
+
+				return $false;
+			},
+			10,
+			3
+		);
+
+		// Force WP_Error response from as http requests are blocked.
+		$this->mock_http_request(
+			$google_proxy->url( 'http://example.com' ),
+			null
+		);
```

Comment:
@benbowler let's better use the real URL for it. You can create another helper function to test errors. Something like this:

```php
private function mock_http_failure( $request_url, $response_error ) {
    add_filter(
        'pre_http_request',
        function( $response, $parsed_args, $url ) use ( $request_url, $response_error ) {
            if ( $url === $request_url ) {
                return new WP_Error( $response_error );
            } else {
                return $response;
            }
        },
        10,
        3
    );
}
```